### PR TITLE
[ci] Fail PRs with the string "no-merge"

### PR DESCRIPTION
**Pull request by ricab**

### CODE REVIEW

**Comment by ricab** [comment]
no-merge just yet

**Comment by ricab** [comment]
Alright this should be ready now.

File: .github/actions/lint/action.yml [hunk 1]
```diff
@@ -4,6 +4,22 @@ description: Run Linting steps
...
+      set -e -o pipefail
+
+      # On pull requests, HEAD^1 will always be the merge base, so diff against that. Look for forbidden strings,
+      # but exclude legitimate cases.
+      forbidden_regex='(?i)\bno[_-]?merge\b'
+      git diff -U0 --no-color HEAD^1 -- . ':(exclude)bors.toml' |
+        (grep -P $forbidden_regex || true) |
+        tee ${HOME}/forbidden_occurrences
+      if [ "$( stat --printf='%s' ${HOME}/forbidden_occurrences )" -ne 0 ]; then
+        echo "##[error] Forbidden string(s) found in the diff 👆 (matched \"$forbidden_regex\")"
```
Comment (Saviq) [inline .github/actions/lint/action.yml:19]: I'd avoid the emoji for a bunch of reasons (colour blindness, missing fonts, etc.).

```suggestion
        echo "##[error] Forbidden string(s) found in the diff ^ (matched \"$forbidden_regex\")"
```

**Review by Saviq — CHANGES_REQUESTED** [review]
Tiny tweak.

File: .github/actions/lint/action.yml (hunk 1 above)
Comment (ricab) [reply .github/actions/lint/action.yml]: OK done.

**Comment by Saviq** [comment]
bors merge

**Review by Saviq — APPROVED** [review]

**Comment by Saviq** [comment]
bors merge

**Comment by ricab** [comment]
@Saviq Codecov is confused so bors won't accept this (and it is not updating the status).

**Comment by Saviq** [comment]
@ricab nah, bors isn't looking at codecov, it's waiting for these now:

https://github.com/canonical/multipass/actions/runs/1854601529

**Comment by ricab** [comment]
Oh indeed, I confused it with the earlier run. Wonder why it's showing that commit's CI as complete:
Screenshot from 2022-02-16 18-58-49

**Comment by ricab** [comment]
bors retry

**Comment by Saviq** [comment]
> Wonder why it's showing that commit's CI as complete:

Because GitHub is meh - if you click on any Details there, you'll see the Integration runs. It probably filters on the `on:` part when showing the dropdown.

**Pull request merged**